lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -1017,26 +985,72 @@ ViewListDiffResult? diffViewList(List<int> active, List<int> next) {
     );
   }
 
-  index = active.lastIndexOf(next.last);
-  if (index != -1) {
+  // This is tried if the last element of the next list is contained in the
+  // active list at `index`. If the lists are in the expected form, we should be
+  // able to iterate backwards from index to the beginning of the active list
+  // and have every element match the corresponding element from the next list.
+  ViewListDiffResult? lookBackwards(int index) {
```

Comment:
Do we need lookBackwards? Can we just do `lookForwards` on `active.reversed`, `next.reversed` and a reversed index (`length - index`?)

(I wonder if `reversed` is a special iterator that starts from the end and does the -- for us, or is it copying the list to reverse it?)

((Also I guess we'd need to also `reverse` before computing the sublists, unless these methods returned the start/end indices of the sublists))